src/python/gem5/components/cachehierarchies/classic/private_l1_private_l2_walk_cache_hierarchy.py

```diff
@@ -57,25 +57,28 @@ def __init__(self, *args, **kwargs) -> None:
 
     @overrides(PrivateL1PrivateL2CacheHierarchy)
     def incorporate_cache(self, board: AbstractBoard) -> None:
-        # ITLB Page walk caches
-        self.iptw_caches = [
-            MMUCache(size="8KiB")
-            for _ in range(board.get_processor().get_num_cores())
-        ]
-        # DTLB Page walk caches
-        self.dptw_caches = [
-            MMUCache(size="8KiB")
-            for _ in range(board.get_processor().get_num_cores())
-        ]
-
+        self._tmp_iptw_caches = []
+        self._tmp_dptw_caches = []
         super().incorporate_cache(board)
-
-        for i, cpu in enumerate(board.get_processor().get_cores()):
-            self.iptw_caches[i].mem_side = self.l2buses[i].cpu_side_ports
-            self.dptw_caches[i].mem_side = self.l2buses[i].cpu_side_ports
+        self.iptw_caches = self._tmp_iptw_caches
+        self.dptw_caches = self._tmp_dptw_caches
+        del self._tmp_iptw_caches
+        del self._tmp_dptw_caches
 
     def _connect_table_walker(self, cpu_id: int, cpu: BaseCPU) -> None:
-        cpu.connect_walker_ports(
-            self.iptw_caches[cpu_id].cpu_side,
-            self.dptw_caches[cpu_id].cpu_side,
-        )
+        walker_ports = cpu.get_mmu().walkerPorts()
+        if len(walker_ports) == 0:
+            return
+
+        dptw_cache = MMUCache(size="8KiB")
+        dptw_cache.mem_side = self.l2buses[cpu_id].cpu_side_ports
+
+        if len(walker_ports) > 1:
```

Comment:
```suggestion
        if len(walker_ports) == 2:
```

I forgot to mention above, but you should probably panic if it's not 0, 1, or 2.